cli/pkg/local/server.go

```diff
@@ -816,3 +819,114 @@ func (nameConflictRetryErrClassifier) Classify(err error) retrier.Action {
 
 	return retrier.Fail
 }
+
+// traceHandler returns trace information corresponding to the trace ID.
+func (s *Server) traceHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		traceID := r.URL.Query().Get("trace_id")
+		resourceName := r.URL.Query().Get("resource_name")
+		if resourceName == "" && traceID == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		if resourceName != "" && traceID != "" {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		db, err := sqlx.Open("duckdb", "")
```

Comment:
Consider moving the logic starting here to `runtime/pkg/observability/fileexporter.go` since it's coupled to the behavior of the file exporter. E.g. it could be `json, err := observability.SearchTracesFile (...)` or something like that. That would also enable having the `traceSpan` type in the `observability` package where it feels like it belongs.